fieldmanager.php

```diff
@@ -499,6 +499,23 @@ function fm_sanitize_textarea( $value ) {
 }
 
 /**
+ * Call wp_die() with an error message for Fieldmanager_Field::_failed_validation().
+ *
+ * @since 1.1.0
+ *
+ * @param string $error_message Optional. Error message in addition to the
+ *                              default message for validation errors.
+ */
+function fm_failed_validation_to_wp_die( $error_message = '' ) {
+	wp_die( esc_html(
+		$error_message
+		. "\n\n"
+		. __( "You may be able to use your browser's back button to resolve this error.", 'fieldmanager' )
+	) );
+}
+add_action( 'fm_failed_validation', 'fm_failed_validation_to_wp_die' );
```

Comment:
2 is definitely better, it's better to collect errors, rely on sanitisation to avoid incorrect data being saved and finally to allow developers to decide what to do in the case of a validation error, most typical use case one I can think of would be keep a post as a draft if the requirements aren't met in that particular context.

A transient is how admin settings errors (added via `add_settings_error()` calls in the `register_setting()` sanitisation callback) are collected and displayed currently so I don't think it's such a bad option. You could also pass data to highlight the affected fields or add an error class.

Also I've implemented validation at the `wp_insert_post_data` hook before to avoid additional calls to `wp_update_post()` but you have to inspect the `$_POST` array rather than looking at post meta that's already been saved.